### PR TITLE
Add support for Origin Access Control CRD

### PR DESCRIPTION
**Pull request by TiberiuGC**
Issue #, if available: https://github.com/aws-controllers-k8s/community/issues/2093

Description of changes: add support for Origin Access Control CRD

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/e2e/tests/test_origin_access_control.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        assert "OriginAccessControlConfig" in latest
+        assert "Description" in latest["OriginAccessControlConfig"]
+        assert (
+            latest["OriginAccessControlConfig"]["Description"]
+            == "a simple origin_access_control"
+        )
+
+        updates = {
+            "spec": {"originAccessControlConfig": {"description": "new description"}},
+        }
```
Comment (a-hilaly) [inline test/e2e/tests/test_origin_access_control.py:109]: We're so lucky we don't have arrays in this CRDs, we don't have write another massacre like https://github.com/aws-controllers-k8s/cloudfront-controller/blob/main/pkg/resource/distribution/hooks.go#L29-L34

**Review by a-hilaly — APPROVED** [review]
Super neat! thanks!
/lgtm

**Comment by bala151187** [comment]
@TiberiuGC @a-hilaly 
**I just installed new verison of cloudfront `0.0.12` & CRD as well .. Facing below permission error for the controller**

`{"level":"info","ts":"2024-07-19T20:56:48.227Z","msg":"pkg/mod/k8s.io/client-go@v0.29.0/tools/cache/reflector.go:229: failed to list *v1alpha1.OriginAccessControl: originaccesscontrols.cloudfront.services.k8s.aws is forbidden: User \"system:serviceaccount:ack-system:ack-cloudfront-controller\" cannot list resource \"originaccesscontrols\" in API group \"cloudfront.services.k8s.aws\" at the cluster scope"}
{"level":"error","ts":"2024-07-19T20:56:48.227Z","msg":"pkg/mod/k8s.io/client-go@v0.29.0/tools/cache/reflector.go:229: Failed to watch *v1alpha1.OriginAccessControl: failed to list *v1alpha1.OriginAccessControl: originaccesscontrols.cloudfront.services.k8s.aws is forbidden: User \"system:serviceaccount:ack-system:ack-cloudfront-controller\" cannot list resource \"originaccesscontrols\" in API group \"cloudfront.services.k8s.aws\" at the cluster scope","stacktrace":"k8s.io/client-go/tools/cache.DefaultWatchErrorHandler\n\t/go/pkg/mod/k8s.io/client-go@v0.29.0/tools/cache/reflector.go:147\nk8s.io/client-go/tools/cache.(*Reflector).Run.func1\n\t/go/pkg/mod/k8s.io/client-go@v0.29.0/tools/cache/reflector.go:292\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\t/go/pkg/mod/k8s.io/apimachinery@v0.29.0/pkg/util/wait/backoff.go:226\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\t/go/pkg/mod/k8s.io/apimachinery@v0.29.0/pkg/util/wait/backoff.go:227\nk8s.io/client-go/tools/cache.(*Reflector).Run\n\t/go/pkg/mod/k8s.io/client-go@v0.29.0/tools/cache/reflector.go:290\nk8s.io/client-go/tools/cache.(*controller).Run.(*Group).StartWithChannel.func2\n\t/go/pkg/mod/k8s.io/apimachinery@v0.29.0/pkg/util/wait/wait.go:55\nk8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1\n\t/go/pkg/mod/k8s.io/apimachinery@v0.29.0/pkg/util/wait/wait.go:72"}`

**my IAM role has these policies attached** 

Screenshot 2024-07-19 at 5 02 19 PM

That role mapped to the service account . 

Screenshot 2024-07-19 at 5 03 53 PM

Am i missing anything

**Comment by a-hilaly** [comment]
@bala151187 this sounds like an RBAC issue, did you update the helm chart as well?

**Comment by bala151187** [comment]
@a-hilaly Yeah . i just manually updated the image :) .

**Comment by a-hilaly** [comment]
Updating the deployment image alone isn't enough, you need to use the latest helm chart (it contains the right RBAC for this version of the controller)

**Comment by bala151187** [comment]
@a-hilaly hmm .. then that's the missing piece .. gotcha .. ll let you know on monday .. thanks for replying

**Pull request merged**